include/thread_pool/thread_pool.hpp

```diff
@@ -169,36 +169,49 @@ inline bool GenericThreadPool<Task, Queue>::tryPost(Handler&& handler)
         auto result = m_idle_workers.tryEmptyAny();
         if (result.first)
         {
-            if (m_workers[result.second]->tryPost(std::forward<Handler>(handler)))
-            {
-                m_workers[result.second]->wake();
-                return true;
-            }
+            auto success = m_workers[result.second]->tryPost(std::forward<Handler>(handler));
+            m_workers[result.second]->wake();
 
-            // If post is unsuccessful, we need to re-add the worker to the idle worker bag.
-            m_idle_workers.fill(result.second);
-            return false;
+            // The above post will only fail if the idle worker's queue is full, which is an extremely 
+            // low probability scenario. In that case, we wake the worker and let it get to work on
+            // processing the items in its queue. We then re-try posting our current task.
+            if (success)
+                return true;
+            else
+                return tryPost(std::forward<Handler>(handler));
```

Comment:
although very low in probability, if tryPost returns false, should we fall thru to the no idle threads code below instead of returning false?